test/integration/test_history_import_export.py

```diff
@@ -198,6 +199,24 @@ def test_export_history_with_discarded_dataset_to_rocrate(self):
         self._wait_for_export_task_on_record(last_record)
         assert last_record["ready"] is True
 
+    def test_export_missing_dataset_fails(self):
+        history_name = f"for_export_failure_{uuid4()}"
+        model_store_format = "rocrate.zip"
+        target_uri = f"gxfiles://posix_test/{history_name}.{model_store_format}"
+        history_id = self.dataset_populator.new_history(history_name)
+        hda = self.dataset_populator.new_dataset(history_id, wait=True)
+
+        # Simulate data loss or some unexpected problem with the dataset file.
+        assert os.path.exists(hda["file_name"])
+        unlink(hda["file_name"])
+        assert not os.path.exists(hda["file_name"])
+
+        # Ensure the export task fails
+        with self.assertRaisesRegex(Exception, "task failed"):
```

Comment:
I suppose we have no way to assert something more specific ?

Reply:
I rewrote the test to use the STS instead of the file sources so we can have a bit more control of the error returned is 
7f5e97a